airflow/models/dag.py

```diff
@@ -800,10 +801,21 @@ def validate(self):
                 f"inconsistent schedule: timetable {self.timetable.summary!r} "
                 f"does not match schedule_interval {self.schedule_interval!r}",
             )
+        self.validate_executor_field()
         self.validate_schedule_and_params()
         self.timetable.validate()
         self.validate_setup_teardown()
 
+    def validate_executor_field(self):
+        for task in self.tasks:
+            if task.executor:
+                try:
+                    ExecutorLoader.lookup_executor_name_by_str(task.executor)
+                except AirflowException:
+                    raise ValueError(
+                        f"The specified executor {task.executor} for task {task.task_id} is not configured"
```

Comment:
```suggestion
                        f"The specified executor {task.executor} for task {task.task_id} is not configured. Review the core.executors Airflow configuration to add it or update the executor configuration for this task."
```

What do you think about this update, which adds some hints to the user on how to rectify the situation?